SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LBDataStorage.cpp

```diff
@@ -4,31 +4,105 @@
+	{
+		switch (Delta.Type)
+		{
+		case EntityDelta::UPDATE:
+		{
+			for (const auto& CompleteUpdate : Delta.ComponentsRefreshed)
+			{
+				for (auto& Storage : DataStorages)
+				{
+					if (Storage->GetComponentsToWatch().Contains(CompleteUpdate.ComponentId))
+					{
+						Storage->OnComponentRefreshed(Delta.EntityId, CompleteUpdate.ComponentId, CompleteUpdate.Data);
+					}
+				}
+			}
+			for (const auto& Update : Delta.ComponentUpdates)
+			{
+				for (auto& Storage : DataStorages)
+				{
+					if (Storage->GetComponentsToWatch().Contains(Update.ComponentId))
+					{
+						Storage->OnUpdate(Delta.EntityId, Update.ComponentId, Update.Update);
+					}
+				}
+			}
+		}
+		break;
+		case EntityDelta::ADD:
+		{
+			const SpatialGDK::EntityViewElement& Element = SubView.GetView().FindChecked(Delta.EntityId);
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnAdded(Delta.EntityId, Element);
+			}
+		}
+		break;
+		case EntityDelta::REMOVE:
+		{
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnRemoved(Delta.EntityId);
+			}
+		}
+		break;
+		case EntityDelta::TEMPORARILY_REMOVED:
+		{
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnRemoved(Delta.EntityId);
+			}
+			const SpatialGDK::EntityViewElement& Element = SubView.GetView().FindChecked(Delta.EntityId);
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnAdded(Delta.EntityId, Element);
+			}
+		}
+		break;
+		default:
+			break;
+		}
+	}
+}
+
+TSet<Worker_ComponentId> FLBDataCollection::GetComponentsToWatch() const
+{
+	TSet<Worker_ComponentId> Components;
+	for (auto Storage : DataStorages)
+	{
+		Components = Components.Union(Storage->GetComponentsToWatch());
+	}
+
+	return Components;
+}
+
 FSpatialPositionStorage::FSpatialPositionStorage()
 {
 	Components.Add(SpatialConstants::POSITION_COMPONENT_ID);
 }
 
-void FSpatialPositionStorage::OnAdded(Worker_EntityId EntityId, const SpatialGDK::EntityViewElement& Element)
+void FSpatialPositionStorage::OnComponentAdded(Worker_EntityId EntityId, Worker_ComponentId ComponentId, Schema_ComponentData* Data)
 {
-	for (const auto& Component : Element.Components)
+	if (ensureAlways(SpatialConstants::POSITION_COMPONENT_ID == ComponentId))
 	{
-		if (SpatialConstants::POSITION_COMPONENT_ID == Component.GetComponentId())
-		{
-			Schema_Object* PositionObject = Schema_GetComponentDataFields(Component.GetUnderlying());
+		Schema_Object* PositionObject = Schema_GetComponentDataFields(Data);
 
-			Schema_Object* CoordsObject = Schema_GetObject(PositionObject, 1);
-			SpatialGDK::Coordinates Coords;
-			Coords.X = Schema_GetDouble(CoordsObject, 1);
-			Coords.Y = Schema_GetDouble(CoordsObject, 2);
-			Coords.Z = Schema_GetDouble(CoordsObject, 3);
+		Schema_Object* CoordsObject = Schema_GetObject(PositionObject, 1);
+		SpatialGDK::Coordinates Coords;
+		Coords.X = Schema_GetDouble(CoordsObject, 1);
+		Coords.Y = Schema_GetDouble(CoordsObject, 2);
+		Coords.Z = Schema_GetDouble(CoordsObject, 3);
```

Comment:
Dedupe this code? Although tbh I'd just add an extra constructor to the `Position` in `StandardLibrary.h` and use it instead.
```suggestion
		SpatialGDK::Coordinates Coords = GetCoordinateFromSchema(PositionObject, 1);
```